example/lib/pages/map_controller.dart

```diff
@@ -187,27 +189,61 @@ class CurrentLocation extends StatefulWidget {
 }
 
 class _CurrentLocationState extends State<CurrentLocation> {
+  static const _eventKey = 'position-key';
+
   var icon = Icons.gps_not_fixed;
+  StreamSubscription<MapEvent> mapEventSubscription;
+
+  @override
+  void dispose() {
+    mapEventSubscription?.cancel();
+    super.dispose();
+  }
+
+  void setIcon(IconData newIcon) {
+    if (newIcon != icon && mounted) {
+      setState(() {
+        icon = newIcon;
+      });
+    }
+  }
 
   void _moveToCurrent() async {
     var location = Location();
 
     try {
+      await mapEventSubscription?.cancel();
+
+      var mapEventStream = widget.mapController.mapEventStream;
+      if (mapEventStream == null) {
+        setIcon(Icons.gps_off);
+        return;
+      }
+
       var currentLocation = await location.getLocation();
-      widget.mapController.move(
-          LatLng(currentLocation.latitude, currentLocation.longitude), 18);
+      var moved = widget.mapController.move(
+        LatLng(currentLocation.latitude, currentLocation.longitude),
+        18,
+        id: _eventKey,
+      );
 
-      setState(() {
-        icon = Icons.gps_fixed;
-      });
-      await widget.mapController.position.first;
-      setState(() {
-        icon = Icons.gps_not_fixed;
-      });
+      if (moved) {
+        setIcon(Icons.gps_fixed);
+
+        mapEventSubscription = mapEventStream.listen((MapEvent event) {
+          if (event is MapEventMove && event.id == _eventKey) {
```

Comment:
It is overkill to listen the stream but here is the example which checks event's id so it's safe to say it's coming from us rather than assume the first move was our one.

Reply:
How can we differentiate between move start and end events? Also, is there a way to dfferantiate among rotate/pan/zoom/double tap zoom? These would be super helpful if you could add them in your PR:D

Reply:
You can check [map_events](https://github.com/fleaflet/flutter_map/blob/a6b5c6f6722e66462b022e49f01f730d68b0d12d/lib/src/gestures/map_events.dart) it is still not documented but you can distinguish events forinstance `MapEventRotateStart` will be fired when map starts to rotate (when `rotationThreshold` is reached the default can be changed in `MapOptions`) and while rotating `MapEventRotate` will be fired continuously and when rotation ends then `MapEventRotateEnd`  will be fired.

Example:
```dart
  @override
  void initState() {
    super.initState();
    mapController = MapController();

    mapController.onReady.then((_) {
      // at this point we can safely listen to [mapEventStream]
      // use stream transformer or anything you want
      subscription = mapController.mapEventStream.listen((MapEvent mapEvent) {
        if (mapEvent is MapEventRotateStart) {
          // do something
        } else if (mapEvent is MapEventFlingEnd ) {
          // do something
        }
      });
    });
  }
```

Reply:
Thanks for the clear example. I didn't see map_events before now it all makes sense. By the way I am a little confused with this scenario:  User can easily both pan zoom at the same time. In this case what events will be emitted?
MapEventMoveStart
MapEventRotateStart
...
MapEventMove
MapEventRotate
MapEventMove
MapEventRotate
MapEventMove
MapEventRotate
...
MapEventMoveEnd
MapEventRotateEnd
Something like this?

Reply:
`Zoom` and `move` are same event. You can read `zoom` which was the zoom when event is `emmited` and there is a `targetZoom` property too. So if `zoom != targetZoom` then zoom has been changed you can compare `center` and `targetCenter` too. Maybe there should be a getter which tells the same so it can be used like this: `if (event is MapEventMove && event.isZoomChanged)`

Reply:
Ok if zoom and move are considered same, let me ask the same question with zoom+rotate? Do we need that extra getter in that case?

Reply:
If only `pinchZoom` and `rotation` are enabled then if only rotation happens then `MapEventMove` won't be fired (but it is almost impossible to do that with fingers).
I think I'll implement zoom threshold too, so rotation and zoom can race. I mean user won't able to rotate and zoom at same time at least Google maps work this way.